simulation/traffic_simulator/src/utils/pose.cpp

```diff
@@ -45,18 +46,31 @@ auto quietNaNLaneletPose() -> traffic_simulator::LaneletPose
 
 auto canonicalize(
   const LaneletPose & lanelet_pose,
-  const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr) -> CanonicalizedLaneletPose
+  const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr)
+  -> std::optional<CanonicalizedLaneletPose>
 {
-  return CanonicalizedLaneletPose(lanelet_pose, hdmap_utils_ptr);
+  if (lanelet_pose == LaneletPose())
+    return std::nullopt;
+  else
+    return CanonicalizedLaneletPose(lanelet_pose, hdmap_utils_ptr);
 }
 
-auto toMapPose(const CanonicalizedLaneletPose & lanelet_pose) -> const geometry_msgs::msg::Pose
+auto toMapPose(const CanonicalizedLaneletPose & lanelet_pose) -> geometry_msgs::msg::Pose
 {
   return static_cast<geometry_msgs::msg::Pose>(lanelet_pose);
 }
 
-auto toLaneletPose(
-  const geometry_msgs::msg::Pose & map_pose, bool include_crosswalk,
+auto toMapPose(
+  const LaneletPose & lanelet_pose,
+  const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr) -> geometry_msgs::msg::Pose
+{
+  return hdmap_utils_ptr
+    ->toMapPose(lanelet_pose, CanonicalizedLaneletPose::getConsiderPoseByRoadSlope())
+    .pose;
+}
```

Comment:
@yamacir-kit 
By “has been completely separated” I mean that now everywhere in traffic_simulator `pose::toMapPose` is used - there is no use of `HdMapUtils::toMapPose` directly anywhere. Namespace `pose::` has free functions that use `HdMapUtils`, some of them are more complex, others are simple calls to function from `HdMapUtils`. 

I've intentionally not moved the implementation from `HdMapUtils::toMapPose` to `pose::toMapPose` at this time, as the goal of this PR is not to refactor `HdMapUtils`. During `HdMapUtils` refactoring (which [I am also in charge of](https://tier4.atlassian.net/browse/RJD-1089)), I would like to split the architecture into `utils` and `lanelet_core` (as in the attached draft image). 
![utils drawio](https://github.com/tier4/scenario_simulator_v2/assets/121798334/65043360-6a2b-495f-a44f-8edaf3c5cf9c)


`HdMapUtils::toMapPose` will probably be moved to `lanelet_core::pose`, since the calculations using `msg::LaneletPose` and `HdMapUtils::getCenterPointsSpline()` are performed there - so it's “near lanelet”

Reply:
https://tier4.atlassian.net/browse/RJD-1089?focusedCommentId=172725

Reply:
@dmoszynski 
I understand. I mark this conversation as resolved.